pkg/kubelet/kubelet.go

```diff
@@ -1575,6 +1585,13 @@ func (kl *Kubelet) syncLoop(updates <-chan PodUpdate, handler SyncHandler) {
 		if err := handler.SyncPods(pods, podSyncTypes, start); err != nil {
 			glog.Errorf("Couldn't sync containers: %v", err)
 		}
+
+		syncedCount++
+		if kl.statusUpdateFrequency != 0 && kl.kubeClient != nil && syncedCount%kl.statusUpdateFrequency == 0 {
```

Comment:
We could run into mod by 0 case, which I don't like. Also one SyncLoop take a long time, but syncedCount is not updated, you might end up missing some NodeStatus update, right? Why not introducing interval for this, and tracking timestamps of last NodeStatus update, and comparing Now to last?